src/NuGet.Services.EndToEnd/Support/Clients/RegistrationClient.cs

```diff
@@ -60,6 +59,61 @@ public RegistrationClient(SimpleHttpClient httpClient, V3IndexClient v3IndexClie
                 logger: logger);
         }
 
+        /// <summary>
+        /// Polls all registration base URLs until the provided ID and version has the specified listed state.
+        /// </summary>
+        /// <param name="id">The package ID.</param>
+        /// <param name="version">The package version.</param>
+        /// <param name="deprecation">The deprecation state to poll for.</param>
+        /// <param name="logger">The logger.</param>
+        /// <returns>Returns a task that completes when the package has the specified deprecation state or the timeout has occurred.</returns>
+        public async Task WaitForDeprecationStateAsync(
+            string id,
+            string version,
+            PackageDeprecationContext deprecation,
+            ITestOutputHelper logger)
+        {
+            var successState = deprecation != null ? "deprecated" : "undeprecated";
+            var failureState = deprecation != null ? "undeprecated" : "deprecated";
+
+            await PollAsync(
+                id,
+                version,
+                semVer2: true,
+                isComplete: catalogEntry => 
+                    catalogEntry.Id == id 
+                    && catalogEntry.Version == version
+                    && (catalogEntry.Deprecation == null) == (deprecation == null)
+                    && HasReason(catalogEntry, "Legacy", deprecation?.IsLegacy ?? false)
+                    && HasReason(catalogEntry, "CriticalBugs", deprecation?.HasCriticalBugs ?? false)
+                    && HasReason(catalogEntry, "Other", deprecation?.IsOther ?? false)
+                    && deprecation?.Message == catalogEntry.Deprecation?.Message 
+                    && deprecation?.AlternatePackageId == catalogEntry.Deprecation?.AlternatePackage?.Id
+                    && catalogEntry.Deprecation?.AlternatePackage?.Range == GetExpectedAlternatePackageRange(deprecation),
+                startingMessage: $"Waiting for package {id} {version} to be {successState} on registration base URLs:",
+                successMessageFormat: $"Package {id} {version} became {successState} on {{0}} after waiting {{1}}.",
+                failureMessageFormat: $"Package {id} {version} was still {failureState} on {{0}} after waiting {{1}}.",
```

Comment:
If one of the deprecation properties (e.g. the message) does not match but the package is still deprecated `failureMessageFormat` will say something confusing.

Reply:
makes sense--changed it